client/js/heat.js

```diff
@@ -0,0 +1,82 @@
+class HeatRow {
+  /**
+   *
+   * @param {int[]} data The array of 0s and 1s for a row in a heat map
+   */
+  constructor(data) {
+    this.data = data;
+  }
+}
+
+// data will be from backend next time
+const TOTAL_LIKED = 13;
+// element value of x means xth latest video is music
+const LIKED_MUSIC_HIST = [1, 4, 7];
+
+// ith value of 1 means ith latest video is music
+const likedMusicBinaryHist = [...Array(TOTAL_LIKED)].fill(0);
```

Comment:
That's a spicy way to create an array filled with zeros. Maybe add a comment that explains that this is what you are doing?